libevmasm/RuleList.h

```diff
@@ -125,50 +127,51 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart2(
-		{{Pattern::Builtins::AND, {{Pattern::Builtins::BYTE, {X, Y}}, {Word(0xff)}}}, [=]() -> Pattern { return {Pattern::Builtins::BYTE, {X, Y}}; }, false},
-		{{Pattern::Builtins::BYTE, {Pattern::WordSize / 8 - 1, X}}, [=]() -> Pattern { return {Pattern::Builtins::AND, {X, Word(0xff)}}; }, false}
+		{Builtins::ADD(X, 0), [=]{ return X; }, false},
+		{Builtins::ADD(0, X), [=]{ return X; }, false},
+		{Builtins::SUB(X, 0), [=]{ return X; }, false},
+		{Builtins::SUB(~Word(0), X), [=]() -> Pattern { return Builtins::NOT(X); }, false},
+		{Builtins::MUL(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::MUL(0, X), [=]{ return Word(0); }, true},
+		{Builtins::MUL(X, 1), [=]{ return X; }, false},
+		{Builtins::MUL(1, X), [=]{ return X; }, false},
+		{Builtins::MUL(X, Word(-1)), [=]() -> Pattern { return Builtins::SUB(0, X); }, false},
+		{Builtins::MUL(Word(-1), X), [=]() -> Pattern { return Builtins::SUB(0, X); }, false},
+		{Builtins::DIV(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::DIV(0, X), [=]{ return Word(0); }, true},
+		{Builtins::DIV(X, 1), [=]{ return X; }, false},
+		{Builtins::SDIV(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::SDIV(0, X), [=]{ return Word(0); }, true},
+		{Builtins::SDIV(X, 1), [=]{ return X; }, false},
+		{Builtins::AND(X, ~Word(0)), [=]{ return X; }, false},
+		{Builtins::AND(~Word(0), X), [=]{ return X; }, false},
+		{Builtins::AND(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::AND(0, X), [=]{ return Word(0); }, true},
+		{Builtins::OR(X, 0), [=]{ return X; }, false},
+		{Builtins::OR(0, X), [=]{ return X; }, false},
+		{Builtins::OR(X, ~Word(0)), [=]{ return ~Word(0); }, true},
+		{Builtins::OR(~Word(0), X), [=]{ return ~Word(0); }, true},
+		{Builtins::XOR(X, 0), [=]{ return X; }, false},
+		{Builtins::XOR(0, X), [=]{ return X; }, false},
+		{Builtins::MOD(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::MOD(0, X), [=]{ return Word(0); }, true},
+		{Builtins::EQ(X, 0), [=]() -> Pattern { return Builtins::ISZERO(X); }, false },
+		{Builtins::EQ(0, X), [=]() -> Pattern { return Builtins::ISZERO(X); }, false },
+		{Builtins::SHL(0, X), [=]{ return X; }, false},
+		{Builtins::SHR(0, X), [=]{ return X; }, false},
+		{Builtins::SHL(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::SHR(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::GT(X, 0), [=]() -> Pattern { return Builtins::ISZERO(Builtins::ISZERO(X)); }, false},
+		{Builtins::LT(0, X), [=]() -> Pattern { return Builtins::ISZERO(Builtins::ISZERO(X)); }, false},
+		{Builtins::GT(X, ~Word(0)), [=]{ return Word(0); }, true},
+		{Builtins::LT(~Word(0), X), [=]{ return Word(0); }, true},
+		{Builtins::GT(0, X), [=]{ return Word(0); }, true},
+		{Builtins::LT(X, 0), [=]{ return Word(0); }, true},
+		{Builtins::AND(Builtins::BYTE(X, Y), Word(0xff)), [=]() -> Pattern { return Builtins::BYTE(X, Y); }, false},
+		{Builtins::BYTE(Pattern::WordSize / 8 - 1, X), [=]() -> Pattern { return Builtins::AND(X, Word(0xff)); }, false}
```

Comment:
```suggestion
		{Builtins::BYTE(Word(Pattern::WordSize / 8 - 1), X), [=]() -> Pattern { return Builtins::AND(X, Word(0xff)); }, false}
```
Just to see if that's it?